middleware/compress.go

```diff
@@ -211,6 +209,8 @@ func (w *maybeCompressResponseWriter) WriteHeader(code int) {
 		if wr := w.encoder(w.ResponseWriter, w.level); wr != nil {
 			w.w = wr
 			w.Header().Set("Content-Encoding", w.encoding)
+			// The content-length after compression is unknown
+			w.Header().Del("Content-Length")			
```

Comment:
I noticed you changed 

```diff
- w.ResponseWriter.Header()
+ w.Header()
```

But I see the other form in this function in other places. Can you fix it all?